edk2toollib/uefi/edk2/parsers/base_parser.py

```diff
@@ -438,60 +482,90 @@ def EvaluateConditional(self, text):
+                found_not = False
+            else:
+                raise RuntimeError(f"Confusing not in tokens: {text}")
+
+        return collapsed_tokens
+
+    @classmethod
+    def _ConvertTokensToPostFix(cls, tokens):
+        # convert infix into post fix
         stack = ["("]
         tokens.append(")")  # add an extra parathesis
         expression = []
         for token in tokens:
+            # If the incoming symbol is a left parenthesis, push it on the stack.
             if token == "(":
                 stack.append(token)
+            # If the incoming symbol is a right parenthesis,
+            # pop the stack and print the operators until you see a left parenthesis.
+            # Discard the pair of parentheses.
             elif token == ")":
                 while len(stack) > 0 and stack[-1] != '(':
                     expression.append(stack.pop())
-            elif token in operators:
-                while len(stack) > 0 and stack[-1] != '(':
-                    self.Logger.debug(stack[-1])
+                stack.pop()  # pop the last (
+            # If this isn't a operator ignore it
+            elif not cls._IsOperator(token):
+                expression.append(token)
+            # If the stack is empty or contains a left parenthesis on top, push the incoming operator onto the stack.
+            elif len(stack) == 0 or stack[-1] == '(':
+                stack.append(token)
+            # If the incoming symbol has higher precedence than the top of the stack, push it on the stack.
+            elif len(stack) == 0 or cls._GetOperatorPrecedence(token) > cls._GetOperatorPrecedence(stack[-1]):
+                stack.append(token)
+            # If the incoming symbol has equal precedence with the top of the stack, use association.
+            # If the association is left to right, pop and print the top of the stack and
+            # then push the incoming operator. If the association is right to left, push the incoming operator.
+            elif len(stack) != 0 and cls._GetOperatorPrecedence(token) == cls._GetOperatorPrecedence(stack[-1]):
+                expression.append(stack.pop())
+                stack.append(token)
+            # If the incoming symbol has lower precedence than the symbol on the top of the stack,
+            # pop the stack and print the top operator.
+            # Then test the incoming operator against the new top of stack.
+            elif len(stack) != 0 and cls._GetOperatorPrecedence(token) < cls._GetOperatorPrecedence(stack[-1]):
+                while len(stack) > 0 and cls._GetOperatorPrecedence(token) <= cls._GetOperatorPrecedence(stack[-1]):
                     expression.append(stack.pop())
                 stack.append(token)
             else:
-                expression.append(token)
+                logging.error("We don't know what to do with " + token)
+            logging.debug(f"{token.ljust(10)} | {' '.join(stack).ljust(50)} | {' '.join(expression).ljust(50)}")
```

Comment:
what is "50".  
Can you add a comment for the magic number usage

Reply:
That's just padding it to a new tabular format. It's pretty arbitrary.

Reply:
I might just take this out